keps/sig-network/20190920-external-dns.md

```diff
@@ -0,0 +1,156 @@
+
+* Is the de facto solution to create DNS records for several Kubernetes resources.
+
+* Is a vital component to achieve an experience close to a PaaS that many Kubernetes users try to replicate on top of Kubernetes, by allowing to automatically create DNS records for web applications.
+
+* Supports already 18 different DNS providers including all major public clouds (AWS, Azure, GCP).
+
+Given that the kubernetes-incubator organization will eventually be shut down, the possible alternatives to moving to be an official Kubernetes project are the following:
+
+* Shut down the project
+
+* Move the project elsewhere
+
+We believe that those alternatives would result in a worse outcome for the community compared to moving the project to the any of the other official Kubernetes organizations.
+In fact, shutting down ExternalDNS can cause:
+
+* The community to rebuild the same solution as already happened multiple times before the project was launched. Currently ExternalDNS is easy to be found, referenced in many articles/tutorials and for that reason not exposed to that risk.
+
+* Existing users of the projects to be left without a future proof working solution.
+
+Moving the ExternalDNS project outside of Kubernetes projects would cause:
+
+* Problems (re-)establishing user trust which could eventually lead to fragmentation and duplication.
+
+* It would be hard to establish in which organization the project should be moved to. The most natural would be Zalando’s organization, being the company that put most of the work on the project. While it is possible to assume Zalando’s commitment to open-source, that would be a strategic mistake for the project community and for the Kubernetes ecosystem due to the obvious lack of neutrality.
+
+* Lack of resources to test, lack of issue management via automation.
+
+For those reasons, we propose to move ExternalDNS out of the Kubernetes incubator, to live either under the kubernetes or kubernetes-sigs organization to keep being a vital part of the Kubernetes ecosystem.
+
+
+## Details
+
+### Graduation Criteria
+
+ExternalDNS is a two years old project widely used in production by many companies. The implementation for the three major cloud providers (AWS, Azure, GCP) is stable, not changing its logic and the project is being used in production by many company using Kubernetes.
+
+We have evidence that many companies are using ExternalDNS in production, but it is out of scope for this proposal to collect a comprehensive list of companies.
+
+The project was quoted by a number of tutorials on the web, including the [official tutorials from AWS](https://aws.amazon.com/blogs/opensource/unified-service-discovery-ecs-kubernetes/).
+
+ExternalDNS can’t be consider to be "done": while the core functionality has been implemented, there is lack of integration testing and structural changes that are needed.
+
+Those are identified in the project roadmap, which is roughly made of the following items:
+
```

Comment:
Can we add fixing some of the release build/artifact related issues to the roadmap?

Reply:
Specifically, having an official k8s.io release image, etc

Reply:
Addressed